wpt cookie path tests

g-k opened the pull request:
Adds wpt tests for the cookie path attribute. 

Notes:
- I included polyfills for fetch and promise to get tests running.  Assuming these changes are OK, I'll remove them and add failure expectations so they don't get synced to w3c.
- I didn't work out a way to test sending cookies when the request and cookie paths exactly match (e.g. request path `/cookies/path/match-exact-page.sub.html` and cookie path `/cookies/path/match-exact-page.sub.html` will not send cookies to `/cookies/resources/echo-json.py`) 
- test names could be clearer
- everything can be squashed to one commit

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #8756 (github issue number if applicable).
- [x] These changes do not require tests because they are tests (and if we test our tests do we need to test our test tests? ...)

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/11624)

<!-- Reviewable:end -->

KiChjang commented:
Fails tidy:
```
--- a/tests/wpt/metadata/MANIFEST.json
+++ b/tests/wpt/metadata/MANIFEST.json
@@ -36030,18 +36030,18 @@
     "deleted_reftests": {},
     "items": {
       "testharness": {
-        "cookies/path/match-exact-slash.sub.html": [
-          {
-            "path": "cookies/path/match-exact-slash.sub.html",
-            "url": "/cookies/path/match-exact-slash.sub.html"
-          }
-        ],
         "cookies/path/match-exact-page.sub.html": [
           {
             "path": "cookies/path/match-exact-page.sub.html",
             "url": "/cookies/path/match-exact-page.sub.html"
           }
         ],
+        "cookies/path/match-exact-slash.sub.html": [
+          {
+            "path": "cookies/path/match-exact-slash.sub.html",
+            "url": "/cookies/path/match-exact-slash.sub.html"
+          }
+        ],
         "cookies/path/match-prefix-cookie-trailing-slash.sub.html": [
           {
             "path": "cookies/path/match-prefix-cookie-trailing-slash.sub.html",
```

g-k commented:
Weird, it's passing for me:

```
$ git log -1 --format=oneline
713440ecd0a892afb3a12bd2459b54ae476bfca9 fixup! test cookie path
$ ./mach test-tidy
  Progress: 100% (19/19)
Running the WPT lint...
  Progress: 100% (16/16)
tidy reported no errors.
```

How do I reproduce that?

highfive commented:
New code was committed to pull request.

KiChjang commented:
r? @Ms2ger

jdm commented:
FYI, that's not a tidy check. It's a separate one that runs ./mach update-manifest and verifies the output of git diff.

bors-servo commented:
:umbrella: The latest upstream changes (presumably #11632) made this pull request unmergeable. Please resolve the merge conflicts.

bors-servo commented:
:umbrella: The latest upstream changes (presumably #11548) made this pull request unmergeable. Please resolve the merge conflicts.

bors-servo commented:
:umbrella: The latest upstream changes (presumably #11699) made this pull request unmergeable. Please resolve the merge conflicts.

bors-servo commented:
:umbrella: The latest upstream changes (presumably #11934) made this pull request unmergeable. Please resolve the merge conflicts.

bors-servo commented:
:umbrella: The latest upstream changes (presumably #12265) made this pull request unmergeable. Please resolve the merge conflicts.

jdm commented:
I'm sorry that this PR kept getting put aside! This is a useful collection of tests that you've assembled, so thank you for the effort you've put into it so far. I've left some comments about how we can make them more maintainable, and echo-json.py and testharness-helpers.js are missing  as well.
-S-awaiting-review  +S-needs-code-changes 

---

Reviewed 17 of 17 files at r1.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[tests/wpt/web-platform-tests/cookies/path/match-exact-page.sub.html, line 5 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11624#-KMeYB2m1rHYj0HhNoQn:-KMeYB2m1rHYj0HhNoQo:-1589030556) ([raw file](https://github.com/servo/servo/blob/c733bd6dac2e8c2a5e756b00f57e3880ec3ef3e6/tests/wpt/web-platform-tests/cookies/path/match-exact-page.sub.html#L5)):*
> ```HTML
> <head>
>   <meta charset=utf-8>
>   <title>Sets cookie from `Set-Cookie` HTTP header for path=/cookies/path/match-exact-page.sub.html</title>
> ```

How about "Set-Cookie header with exact path match" instead?

---

*[tests/wpt/web-platform-tests/cookies/path/match-exact-page.sub.html, line 21 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11624#-KMeY37acGq0I7v4wSMR:-KMeY37bPKG7Xk_-FSRl:-1595731180) ([raw file](https://github.com/servo/servo/blob/c733bd6dac2e8c2a5e756b00f57e3880ec3ef3e6/tests/wpt/web-platform-tests/cookies/path/match-exact-page.sub.html#L21)):*
> ```HTML
>   test(function () {
>       assert_not_equals(document.cookie.match(/path_matches=1/), null);
>   }, "'path_matches=1' present in `document.cookie`");
> ```

nit: not present

---

*[tests/wpt/web-platform-tests/cookies/path/match-exact-page.sub.html, line 23 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11624#-KMeXzlfpd_gBHZFYP0U:-KMeXzlfpd_gBHZFYP0V:-409686441) ([raw file](https://github.com/servo/servo/blob/c733bd6dac2e8c2a5e756b00f57e3880ec3ef3e6/tests/wpt/web-platform-tests/cookies/path/match-exact-page.sub.html#L23)):*
> ```HTML
>   }, "'path_matches=1' present in `document.cookie`");
> 
>   // TODO: make sure the cookie is sent and echoed from /cookies/path/match-exact-page.sub.html
> ```

It looks like this test is unfinished?

---

*[tests/wpt/web-platform-tests/cookies/path/match-exact-page.sub.html.headers, line 4 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11624#-KMeYUkL6Qi2eaCUeY4P:-KMeYUkL6Qi2eaCUeY4Q:4763127) ([raw file](https://github.com/servo/servo/blob/c733bd6dac2e8c2a5e756b00f57e3880ec3ef3e6/tests/wpt/web-platform-tests/cookies/path/match-exact-page.sub.html.headers#L4)):*
> ```
> Cache-Control: no-store, no-cache, must-revalidate
> Cache-Control: post-check=0, pre-check=0, false
> Pragma: no-cache
> ```

Are all these cache-related headers necessary?

---

*[tests/wpt/web-platform-tests/cookies/path/match-exact-slash.sub.html, line 23 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11624#-KMeYf-456yb-t_y9FRt:-KMeYf-456yb-t_y9FRu:-1595731180) ([raw file](https://github.com/servo/servo/blob/c733bd6dac2e8c2a5e756b00f57e3880ec3ef3e6/tests/wpt/web-platform-tests/cookies/path/match-exact-slash.sub.html#L23)):*
> ```HTML
>   test(function () {
>       assert_not_equals(document.cookie.match(/path_matches=1/), null);
>   }, "'path_matches=1' present in `document.cookie`");
> ```

nit: not present

---

*[tests/wpt/web-platform-tests/cookies/path/match-exact-slash.sub.html, line 28 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11624#-KMeYsg4bykDZMZdpmWs:-KMeYsg5IoM1uQRTFhH-:-649767229) ([raw file](https://github.com/servo/servo/blob/c733bd6dac2e8c2a5e756b00f57e3880ec3ef3e6/tests/wpt/web-platform-tests/cookies/path/match-exact-slash.sub.html#L28)):*
> ```HTML
>       t.add_cleanup(clearKnownCookie);
> 
>       return fetch("http://{{host}}:{{ports[http][0]}}/cookies/resources/echo-json.py",
> ```

Isn't a URL of "/cookies/resources/echo-json.py" equivalent? It's better to avoid using absolute URLs when possible.

---

*[tests/wpt/web-platform-tests/cookies/path/match-prefix-request-slash.sub.html, line 1 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11624#-KMet9X2o5VYaoE4AQ3H:-KMet9X32Bv6rtpLnSOr:-973311782) ([raw file](https://github.com/servo/servo/blob/c733bd6dac2e8c2a5e756b00f57e3880ec3ef3e6/tests/wpt/web-platform-tests/cookies/path/match-prefix-request-slash.sub.html#L1)):*
> ```HTML
> <!doctype html>
> ```

Given how similar all of the tests for match paths are, I think we should combine them into a single test file that executes a series of tests sequentially - load an iframe from a python script that allows the test to specify the desired path value, then have the iframe report its `document.cookie` value, then clear the cookie and proceed to the next test.

---

*[tests/wpt/web-platform-tests/cookies/path/no-match-exact.sub.html, line 28 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11624#-KMeuFwlf0VWUTZhfVKD:-KMeuFwlf0VWUTZhfVKE:-1502100780) ([raw file](https://github.com/servo/servo/blob/c733bd6dac2e8c2a5e756b00f57e3880ec3ef3e6/tests/wpt/web-platform-tests/cookies/path/no-match-exact.sub.html#L28)):*
> ```HTML
>       t.add_cleanup(clearKnownCookie);
> 
>       document.cookie = "path_matches=1; expires=Thu, 01 Jan 1970 00:00:01 GMT; path=/w/";
> ```

We should use a separate cookie name here so we test both the HTTP and DOM code paths.

---

*[tests/wpt/web-platform-tests/cookies/path/no-match-exact.sub.html, line 30 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11624#-KMetx1PaBAnETsGaAPl:-KMetx1Q_8dY4bHkaZtb:1460650765) ([raw file](https://github.com/servo/servo/blob/c733bd6dac2e8c2a5e756b00f57e3880ec3ef3e6/tests/wpt/web-platform-tests/cookies/path/no-match-exact.sub.html#L30)):*
> ```HTML
>       document.cookie = "path_matches=1; expires=Thu, 01 Jan 1970 00:00:01 GMT; path=/w/";
> 
>       return fetch("http://{{host}}:{{ports[http][0]}}/cookies/resources/echo-json.py",
> ```

We should be able to use a relative URL here.

---

*[tests/wpt/web-platform-tests/cookies/path/no-match-no-cookie-trailing-slash.sub.html, line 1 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11624#-KMeuTwtdivL0XRSC2Qm:-KMeuTwtdivL0XRSC2Qn:301515773) ([raw file](https://github.com/servo/servo/blob/c733bd6dac2e8c2a5e756b00f57e3880ec3ef3e6/tests/wpt/web-platform-tests/cookies/path/no-match-no-cookie-trailing-slash.sub.html#L1)):*
> ```HTML
> <!doctype html>
> ```

We should be able to combine these negative tests into one file as previously described.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11624#-:-KMesOA1q99uJ-PA4fZO:-692628165)*
<!-- Sent from Reviewable.io -->

jdm commented:
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 18 of 18 files at r3.
Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[tests/wpt/web-platform-tests/cookies/path/echo-cookie.html, line 15 \[r3\]](https://reviewable.io:443/reviews/servo/servo/11624#-KNXOg8Em5cmtZSfW9YC:-KNXOg8Em5cmtZSfW9YD:1560216344) ([raw file](https://github.com/servo/servo/blob/56ac3d5e6dabbc73b65ded5f67efbee910291d7a/tests/wpt/web-platform-tests/cookies/path/echo-cookie.html#L15)):*
> ```HTML
>  // console.log("path", window.location.pathname);
>  // console.log("have cookies", document.cookie);
>  // console.log("parent", window.parent.foo(2));
> ```

These and other commented code will need to be removed.

---

*[tests/wpt/web-platform-tests/cookies/path/echo-cookie.html, line 27 \[r3\]](https://reviewable.io:443/reviews/servo/servo/11624#-KNXOodCzb30UIPHKZuR:-KNXOodDMXzrmi99xs9X:360750943) ([raw file](https://github.com/servo/servo/blob/56ac3d5e6dabbc73b65ded5f67efbee910291d7a/tests/wpt/web-platform-tests/cookies/path/echo-cookie.html#L27)):*
> ```HTML
>  window.canSetCookieFromHeader = function (name, path) {
>      // fetch("/cookies/resources/set-cookie.py", {"name": name, "path": path});
>      return "ok";
> ```

What's going on here?

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 18 \[r3\]](https://reviewable.io:443/reviews/servo/servo/11624#-KNXPA8BzPoVgXH4Uudk:-KNXPA8BzPoVgXH4Uudl:604682979) ([raw file](https://github.com/servo/servo/blob/56ac3d5e6dabbc73b65ded5f67efbee910291d7a/tests/wpt/web-platform-tests/cookies/path/match.html#L18)):*
> ```HTML
>      var body = document.getElementsByTagName('body')[0];
>      var createIframe = function createIframe(callback) {
> 	 var iframe = document.createElement('iframe');
> ```

nit: the tabs in this file should be replaced with spaces.

---

*[tests/wpt/web-platform-tests/cookies/path/set-cookie.py, line 3 \[r3\]](https://reviewable.io:443/reviews/servo/servo/11624#-KNXPMTk7l2vzvdJhR7c:-KNXPMTk7l2vzvdJhR7d:919338253) ([raw file](https://github.com/servo/servo/blob/56ac3d5e6dabbc73b65ded5f67efbee910291d7a/tests/wpt/web-platform-tests/cookies/path/set-cookie.py#L3)):*
> ```Python
> import json
> 
> # TODO: when given a cookie name and path it should return them in a Set-Cookie header
> ```

Clearly there's work to be done here; just leaving a reminder for myself as well.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11624#-:-KNXaV3TJb2X-JOijI9R:1458266033)*
<!-- Sent from Reviewable.io -->

bors-servo commented:
:umbrella: The latest upstream changes (presumably #12598) made this pull request unmergeable. Please resolve the merge conflicts.

jdm commented:
-S-needs-rebase -S-needs-rebase +S-needs-code-changes 

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11624#-:-KTFxVKm484PXjhMaZv9:b-bsb3nt)*
<!-- Sent from Reviewable.io -->

g-k committed:
test cookie path matching

jdm commented:
This is looking really great! I'm very interested in that uncaught exception you saw.
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 9 of 21 files at r5.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[tests/wpt/web-platform-tests/cookies/path/echo-cookie.html, line 13 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGQPjD4CeTJG-Z9adA:-KTGQPjE7LO2amH2Eups:bbu6vf0) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/path/echo-cookie.html#L13)):*
> ```HTML
>   document.cookie = name + '=1; path = ' + path + ';';
> }
> window.fetchCookieThen = function (name, path) {
> ```

The name implies a callback or returning a promise or something. 

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 41 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGRt8RAUwrjqYqKkiR:-KTGRt8SkuTLMXTSasqD:bhyuzvs) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/path/match.html#L41)):*
> ```HTML
>   var iframe = createIframeThen(function () {
>     iframe.contentWindow.fetchCookieThen(testCase.name, testCase.path);
>       // TODO: figure out 'ERROR:script::dom::bindings::error: Uncaught exception: failed to extract information' error
> ```

Spooky: https://dxr.mozilla.org/servo/source/components/script/dom/bindings/error.rs?q=failed+to+extract&redirect_type=single#221 . If you have a way to reproduce this, that would be really useful.

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 42 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGRz92VfCVxlw0OVOk:-KTGRz92VfCVxlw0OVOl:bqupetk) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/path/match.html#L42)):*
> ```HTML
>     iframe.contentWindow.fetchCookieThen(testCase.name, testCase.path);
>       // TODO: figure out 'ERROR:script::dom::bindings::error: Uncaught exception: failed to extract information' error
>       window.setTimeout(function (testCase, ctx, iframe) {
> ```

We'll need to solve the previous problem and remove this setTimeout before we can merge this.

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 50 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGQ_JOIzSYDG_bbbz2:-KTGQ_JOIzSYDG_bbbz3:b94ynum) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/path/match.html#L50)):*
> ```HTML
> 	  }
> 	  iframe.contentWindow.expireCookie(testCase.name, testCase.path);
> 	  ctx.done();
> ```

nit: spaces instead of tabs.

---

*[tests/wpt/web-platform-tests/cookies/resources/set-cookie.py, line 24 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGQezt3xldmRbpuiwf:-KTGQezuXyqx0FpU8SRc:bsm0okv) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/resources/set-cookie.py#L24)):*
> ```Python
>     """
>     params = urlparse.parse_qs(request.url_parts.query)
>     print >> sys.stderr, params
> ```

Let's remove the print statements in this file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11624#-:-KTGStYIXyuI44LTu8a0:b-a5n78b)*
<!-- Sent from Reviewable.io -->

g-k commented:
Review status: 1 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[tests/wpt/web-platform-tests/cookies/path/echo-cookie.html, line 13 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGQPjD4CeTJG-Z9adA:-KTH4DX1ACLSKSd5VQME:bbkfmml) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/path/echo-cookie.html#L13)):*
<details><summary><i>Previously, jdm (Josh Matthews) wrote…</i></summary>
> The name implies a callback or returning a promise or something. 
</details>
Switched it back to returning a  promise.

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 41 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGRt8RAUwrjqYqKkiR:-KTH11W4gtV_wYs2QDCh:b-7v0n2) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/path/match.html#L41)):*
<details><summary><i>Previously, jdm (Josh Matthews) wrote…</i></summary>
> Spooky: https://dxr.mozilla.org/servo/source/components/script/dom/bindings/error.rs?q=failed+to+extract&redirect_type=single#221 . If you have a way to reproduce this, that would be really useful.
</details>
I pushed: https://github.com/g-k/servo/tree/failed-to-extract-info-js-error
and can reproduce it there but haven't pared it down.

Calling executeTestsSerially directly (without the tests.map call) and removing the iframe made the error go away. 

Maybe something to do with the JS VM losing stack info when the iframe promise continues in another window?

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 42 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGRz92VfCVxlw0OVOk:-KTH2Tt7KboSVOW3mo5u:byztyva) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/path/match.html#L42)):*
<details><summary><i>Previously, jdm (Josh Matthews) wrote…</i></summary>
> We'll need to solve the previous problem and remove this setTimeout before we can merge this.
</details>
Fixed on the most recent push.

The fetch API calls or server response isn't setting cookies (tried checking with an alert in https://github.com/servo/servo/pull/11624/commits/e16796880fa4c819d5f58a9f4a74593aa0963802) and the assertion failure triggers the promise `catch` (for some reason I thought the assertion would be self contained and it looks like promise_test does that).

This was the result of trying a few different things like not using an iframe, not passing the callback to the iframe, using post message instead, XHRs instead of the fetch API, etc.

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 50 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGQ_JOIzSYDG_bbbz2:-KTH3vL6D7rcQrxBqIPd:bhnd92p) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/path/match.html#L50)):*
<details><summary><i>Previously, jdm (Josh Matthews) wrote…</i></summary>
> nit: spaces instead of tabs.
</details>
Should be fixed or at least

grep '\t' tests/wpt/web-platform-tests/cookies/path/match.html

doesn't return anything.

---

*[tests/wpt/web-platform-tests/cookies/resources/set-cookie.py, line 24 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGQezt3xldmRbpuiwf:-KTH4tjfX2T9rmfAhNo0:b-vys64n) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/resources/set-cookie.py#L24)):*
<details><summary><i>Previously, jdm (Josh Matthews) wrote…</i></summary>
> Let's remove the print statements in this file.
</details>
Done.  https://github.com/servo/servo/pull/11624/commits/6fcbcf8cf1c840f82312dba3a62620a19ed8027e

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11624)*
<!-- Sent from Reviewable.io -->

jdm commented:
Almost there! A couple last fixups to maximize the ability to report errors effectively and this should be ready to merge!
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 41 at r5](https://reviewable.io:443/reviews/servo/servo/11624#-KTGRt8RAUwrjqYqKkiR:-KTyHE66GkW2BYckqz6n:b7oiyyw) ([raw file](https://github.com/servo/servo/blob/4ed7305029bc506ddfbb0752d1b6ce4d8fec7648/tests/wpt/web-platform-tests/cookies/path/match.html#L41)):*
<details><summary><i>Previously, g-k (Greg Guthe) wrote…</i></summary>
> I pushed: https://github.com/g-k/servo/tree/failed-to-extract-info-js-error
> and can reproduce it there but haven't pared it down.
> 
> Calling executeTestsSerially directly (without the tests.map call) and removing the iframe made the error go away. 
> 
> Maybe something to do with the JS VM losing stack info when the iframe promise continues in another window?
</details>
Thanks! I filed #13748 based on my investigations of this.

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 25 at r6](https://reviewable.io:443/reviews/servo/servo/11624#-KTy81CnplPunOnAtglT:-KTy81Coc7cNFCJYhHti:bz7iffz) ([raw file](https://github.com/servo/servo/blob/93b13d09fc9eb99cb21d58895699a705f227f7de/tests/wpt/web-platform-tests/cookies/path/match.html#L25)):*
> ```HTML
> };
> var testCookiePathFromDOM = function (testCase, test) {
>   var iframe = createIframeThen(function () {
> ```

`createIframeThen(test.step_func(function() {`

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 39 at r6](https://reviewable.io:443/reviews/servo/servo/11624#-KTy85vcqF4DtWD_QFRv:-KTy85vcqF4DtWD_QFRw:bz7iffz) ([raw file](https://github.com/servo/servo/blob/93b13d09fc9eb99cb21d58895699a705f227f7de/tests/wpt/web-platform-tests/cookies/path/match.html#L39)):*
> ```HTML
> };
> var testCookiePathFromHeader = function (testCase, test) {
>   var iframe = createIframeThen(function () {
> ```

`createIframeThen(test.step_func(function() {`

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 40 at r6](https://reviewable.io:443/reviews/servo/servo/11624#-KTy8EPLXdhHfYH8MrPm:-KTy8EPLXdhHfYH8MrPn:b-iw70so) ([raw file](https://github.com/servo/servo/blob/93b13d09fc9eb99cb21d58895699a705f227f7de/tests/wpt/web-platform-tests/cookies/path/match.html#L40)):*
> ```HTML
> var testCookiePathFromHeader = function (testCase, test) {
>   var iframe = createIframeThen(function () {
>     iframe.contentWindow.fetchCookieThen(testCase.name, testCase.path).then(function (response) {
> ```

`.then(test.step_func(function(response) {`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11624#-:-KTyHRiZVxwehWz5d3wf:bokhsak)*
<!-- Sent from Reviewable.io -->

g-k commented:
Review status: 2 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 25 at r6](https://reviewable.io:443/reviews/servo/servo/11624#-KTy81CnplPunOnAtglT:-KU4nwUgNSG6yFwDA4WS:b-896fix) ([raw file](https://github.com/servo/servo/blob/93b13d09fc9eb99cb21d58895699a705f227f7de/tests/wpt/web-platform-tests/cookies/path/match.html#L25)):*
<details><summary><i>Previously, jdm (Josh Matthews) wrote…</i></summary>
> `createIframeThen(test.step_func(function() {`
</details>
Done.

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 39 at r6](https://reviewable.io:443/reviews/servo/servo/11624#-KTy85vcqF4DtWD_QFRv:-KU4nx8w1-s0uHQ86KFI:b-896fix) ([raw file](https://github.com/servo/servo/blob/93b13d09fc9eb99cb21d58895699a705f227f7de/tests/wpt/web-platform-tests/cookies/path/match.html#L39)):*
<details><summary><i>Previously, jdm (Josh Matthews) wrote…</i></summary>
> `createIframeThen(test.step_func(function() {`
</details>
Done.

---

*[tests/wpt/web-platform-tests/cookies/path/match.html, line 40 at r6](https://reviewable.io:443/reviews/servo/servo/11624#-KTy8EPLXdhHfYH8MrPm:-KU4nxXK6_FN1fnGVlbp:b-896fix) ([raw file](https://github.com/servo/servo/blob/93b13d09fc9eb99cb21d58895699a705f227f7de/tests/wpt/web-platform-tests/cookies/path/match.html#L40)):*
<details><summary><i>Previously, jdm (Josh Matthews) wrote…</i></summary>
> `.then(test.step_func(function(response) {`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11624)*
<!-- Sent from Reviewable.io -->

jdm commented:
@bors-servo: r+
Woo!

bors-servo commented:
:pushpin: Commit ec68a99 has been approved by `jdm`

<!-- @bors-servo r=jdm ec68a990d03444ec7ceabef212c0233c28c7f5d5 -->

bors-servo commented:
:hourglass: Testing commit ec68a99 with merge ecbdeb3...

bors-servo commented:
:sunny: Test successful - [arm32](http://build.servo.org/builders/arm32/builds/3556), [arm64](http://build.servo.org/builders/arm64/builds/3567), [linux-dev](http://build.servo.org/builders/linux-dev/builds/3633), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/264), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/265), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/3634), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/3612), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/314), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/316), [windows-dev](http://build.servo.org/builders/windows-dev/builds/1523)
